converters/toZigbee.js

```diff
@@ -2004,95 +2004,140 @@ const converters = {
-            const payload = {0x4030: {value: (value ? 0x01: 0x00), type: 0x10}};
+            const payload = {'danfossHeatAvailable': (value ? 1: 0)};
             await entity.write('hvacThermostat', payload, manufacturerOptions.danfoss);
+            return {readAfterWriteTime: 200, state: {'heat_available': value}};
+        },
+        convertGet: async (entity, key, meta) => {
+            await entity.read('hvacThermostat', ['danfossHeatAvailable'], manufacturerOptions.danfoss);
         },
+    },
+    danfoss_heat_required: {
+        key: ['heat_required'],
         convertGet: async (entity, key, meta) => {
-            await entity.read('hvacThermostat', [0x4030], manufacturerOptions.danfoss);
+            await entity.read('hvacThermostat', ['danfossHeatRequired'], manufacturerOptions.danfoss);
         },
     },
     danfoss_day_of_week: {
         key: ['day_of_week'],
         convertSet: async (entity, key, value, meta) => {
-            const payload = {0x4010: {value: (Math.abs(value) < 7 ? Math.abs(value) : 7), type: 0x30}};
+            const payload = {'danfossDayOfWeek': (Math.abs(value) < 7 ? Math.abs(value) : 7)};
             await entity.write('hvacThermostat', payload, manufacturerOptions.danfoss);
+            return {readAfterWriteTime: 200, state: {'day_of_week': value}};
         },
         convertGet: async (entity, key, meta) => {
-            await entity.read('hvacThermostat', [0x4010], manufacturerOptions.danfoss);
+            await entity.read('hvacThermostat', ['danfossDayOfWeek'], manufacturerOptions.danfoss);
         },
     },
     danfoss_trigger_time: {
         key: ['trigger_time'],
         convertSet: async (entity, key, value, meta) => {
-            const payload = {0x4011: {value: value, type: 0x21}};
+            const payload = {'danfossTriggerTime': value};
             await entity.write('hvacThermostat', payload, manufacturerOptions.danfoss);
+            return {readAfterWriteTime: 200, state: {'trigger_time': value}};
         },
         convertGet: async (entity, key, meta) => {
-            await entity.read('hvacThermostat', [0x4011], manufacturerOptions.danfoss);
+            await entity.read('hvacThermostat', ['danfossTriggerTime'], manufacturerOptions.danfoss);
         },
     },
-    danfoss_window_open: {
+    danfoss_window_open_internal: {
+        key: ['window_open_internal'],
+        convertGet: async (entity, key, meta) => {
+            await entity.read('hvacThermostat', ['danfossWindowOpenInternal'], manufacturerOptions.danfoss);
+        },
+    },
+    danfoss_window_open_external: {
         key: ['window_open_external'],
         convertSet: async (entity, key, value, meta) => {
-            const payload = {0x4003: {value: (value ? 0x01: 0x00), type: 0x10}};
+            const payload = {'danfossWindowOpenExternal': (value ? 1: 0)};
             await entity.write('hvacThermostat', payload, manufacturerOptions.danfoss);
```

Comment:
`await entity.write('hvacThermostat', {'danfossWindowOpenExternal': value}, manufacturerOptions.danfoss);` should be fine too I think, given the type of bool for danfossWindowOpenExternal

Reply:
Solved here
https://github.com/Koenkk/zigbee-herdsman-converters/pull/2592/commits/5338a8efcec58d1018d252f62bb519ace285fefd